File: 1-two-sum/1-two-sum.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
# make a new hash 
 checked = {}
    
    nums.each_with_index do |num,index|
        diff = target - num 
        
        if checked[diff]
            return [checked[diff],index]
        else 
            checked[num] = index
        end 
    end 
end

File: 1-two-sum/1-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
//  const hash_map = {}
 
//  for(let i = 0; i < nums.length; i++){
//     const diff = target - nums[i]
    
//         if(diff in hash_map){
//             return [hash_map[diff], i]
//         }else{
//         hash_map[nums[i]] = i 
//     }
    
const hash = {}
 
 for(let i=0; i <= nums.length; i++){
     const diff = target - nums[i]
     if(diff in hash){
         return [hash[diff], i]
     }
     else{
         hash[nums[i]] = i 
     }
  }
    
    
 }
    
    